File: py/src/args.py
import argparse


def parse_args(args):
    parser = argparse.ArgumentParser()
    parser.add_argument("rom")
    parser.add_argument(
        "--info", action="store_true", default=False, help="Show ROM metadata"
    )
    parser.add_argument("-c", "--debug-cpu", action="store_true", default=False)
    parser.add_argument("-g", "--debug-gpu", action="store_true", default=False)
    parser.add_argument("-r", "--debug-ram", action="store_true", default=False)
    parser.add_argument("-H", "--headless", action="store_true", default=False)
    parser.add_argument("-S", "--silent", action="store_true", default=False)
    parser.add_argument("-t", "--turbo", action="store_true", default=False)
    parser.add_argument(
        "-p",
        "--profile",
        type=int,
        help="Exit after N frames",
        default=0,
        metavar="N",
    )
    return parser.parse_args(args)

File: js/src/args.ts
import { Command, InvalidArgumentError } from "commander";

function myParseInt(value: string, dummyPrevious: number): number {
    const parsedValue = parseInt(value, 10);
    if (isNaN(parsedValue)) {
        throw new InvalidArgumentError("Not a number.");
    }
    return parsedValue;
}

export function parse_args(argv: Array<string>) {
    const program = new Command();
    program
        .version("0.0.0")
        .description("A gameboy emulator in many languages, javascript edition")
        .option("-c, --debug-cpu", "Debug CPU")
        .option("-g, --debug-gpu", "Debug GPU")
        .option("-r, --debug-ram", "Debug RAM")
        .option("-H, --headless", "Disable GUI")
        .option("-S, --silent", "Disable Sound")
        .option("-t, --turbo", "No sleep()")
        .option("-f, --frames <n>", "Exit after this many frames", myParseInt)
        .option("-p, --profile <n>", "Exit after this many seconds", myParseInt)
        .argument("rom", "path to .gb file");

    let parsed = program.parse(argv);
    let args = parsed.opts();
    args.rom = parsed.args[0];
    return args;
}
